yellow-paper/docs/circuits/public-kernel-iterative.md

```diff
@@ -0,0 +1,280 @@
+# Public Kernel Circuit - Iterative
+
+:::info Disclaimer
+This is a draft. These requirements need to be considered by the wider team, and might change significantly before a mainnet release.
+:::
+
+## Requirements
+
+In the public kernel iteration, the process involves taking a previous iteration and public call data, verifying their integrity, and preparing the necessary data for subsequent circuits to operate.
+
+### Verification of the Previous Iteration
+
+#### Verifying the previous kernel proof.
+
+It verifies that the previous iteration was executed successfully with the given proof data, verification key, and public inputs.
+
+The preceding proof can be:
+
+- [tail private kernel proof](./private-kernel-tail.md).
+- Iterative public kernel proof.
+
+### Processing Public Function Call
+
+#### Ensuring the contract instance being called is deployed.
+
+1. The nullifier representing the contract exists in the contract tree.
+
+   - Specifically, this nullifier is the contract address siloed with the address of a precompiled deployment contract.
+
+2. The contract is listed in the new contracts within the public inputs.
+
+   - The index of this contract in the new contracts array is supplied as a hint through private inputs.
+   - The counter of the new contract context must be less than the _counter_start_ of the current call.
+
+#### Ensuring the function being called exists in the contract.
+
+The contract address contains the contract class ID, which is a hash of the root of its function tree and additional values. This circuit leverages these characteristics to establish the validity of the function's association with the contract address.
+
+Each leaf of the function tree is a hash representing a function. The preimage includes:
```

Comment:
We'll need to update this when the public functions are all inlined in the same bytecode blob, but for now this is what we have!